tests/test_mailsyncer.py

```diff
@@ -0,0 +1,257 @@
+	syncer = MailSyncer([])
+	account = FakeAccount()
+	messages = make_messages(0, 10)
+	account.set_current_messages(messages)
+	mails_in_first_sync = syncer.sync([account])
+	new_messages = make_messages(10, 14)
+	account.set_current_messages(messages + new_messages)
+	mails_in_second_sync = syncer.sync([account])
+	assert len(mails_in_second_sync) == 14
+	assert all(mail in mails_in_second_sync for mail in mails_in_first_sync)
+
+
+def test_syncing_should_update_messages():
+	syncer = MailSyncer([])
+	account = FakeAccount()
+	messages = make_messages(0, 10)
+	account.set_current_messages(messages)
+	mails_in_first_sync = syncer.sync([account])
+	new_messages = make_messages(10, 14)
+	account.set_current_messages(messages[3:] + new_messages)
+	mails_in_second_sync = syncer.sync([account])
+	assert len(mails_in_second_sync) == 11
+
+
+def test_syncing_without_account_should_keep_already_synced_messages():
+	syncer = MailSyncer([])
+	account = FakeAccount()
+	messages = make_messages(0, 10)
+	account.set_current_messages(messages)
+	mails_in_first_sync = syncer.sync([account])
+	mails_in_second_sync = syncer.sync([])
+	assert mails_in_first_sync == mails_in_second_sync
+
+
+def test_syncing_multiple_accounts_should_collect_all_messages():
+	syncer = MailSyncer([])
+	account1 = FakeAccount()
+	account2 = FakeAccount()
+	account3 = FakeAccount()
+	account1.set_current_messages(make_messages(0, 10))
+	account2.set_current_messages(make_messages(10, 20))
+	account3.set_current_messages(make_messages(20, 30))
+	mails = syncer.sync([account1, account2, account3])
+	assert len(mails) == 30
+
+
+def test_syncing_multiple_account_separately_should_collect_all_messages():
+	syncer = MailSyncer([])
+	# Note: Accounts are identified by user, server and folders attributes, 
+	#       see Account.get_id()
+	account1 = FakeAccount(name='1', user='a')
+	account2 = FakeAccount(name='2', server='b')
+	account3 = FakeAccount(name='3')
+	account1.set_current_messages(make_messages(0, 10))
+	account2.set_current_messages(make_messages(10, 20))
+	account3.set_current_messages(make_messages(20, 30))
+	syncer.sync([account1])
+	syncer.sync([account2])
+	mails = syncer.sync([account3])
+	assert len(mails) == 30
+
+
+def test_syncing_same_mail_twice_from_same_account():
+	syncer = MailSyncer([])
+	account = FakeAccount()
+	messages1 = make_messages(0, 1)
+	messages2 = make_messages(0, 1)
+	account.set_current_messages(messages1 + messages2)
+	mails = syncer.sync([account])
+	assert len(mails) == 1
+
+
+def test_syncing_same_mail_from_different_accounts():
+	syncer = MailSyncer([])
+	account1 = FakeAccount(name='1', user='a')
+	account2 = FakeAccount(name='2', server='b')
+	messages1 = make_messages(0, 1)
+	messages2 = make_messages(0, 1)
+	account1.set_current_messages(messages1)
+	account2.set_current_messages(messages2)
+	mails = syncer.sync([account1, account2])
+
+	# Note: Is this ok, or should it list message in both accounts?
+	assert len(mails) == 1
```

Comment:
If I send a mail to two addresses and it goes to two different accounts/mailboxes, Mailnag shows the mail only once, since both mails have same message-id.